src/System.Net.WebSockets.Client/tests/SendReceiveTest.cs

```diff
@@ -144,59 +132,34 @@ public class SendReceiveTest : ClientWebSocketTestBase
         {
             using (ClientWebSocket cws = await WebSocketHelper.GetConnectedWebSocket(server, TimeOutMilliseconds, _output))
             {
-                var cts = new CancellationTokenSource(TimeOutMilliseconds);
-
-                Task[] tasks = new Task[2];
-
-                await cws.SendAsync(
-                    WebSocketData.GetBufferFromText(".delay5sec"),
-                    WebSocketMessageType.Text,
-                    true,
-                    cts.Token);
+                var tasks = new Task[2];
```

Comment:
@stephentoub - more information about how this is not a race condition on windows (currently).
The critical part is in [WinHttpWebSocket](https://github.com/dotnet/corefx/blob/master/src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinHttpWebSocket.cs#L395-L446), but what ends up happening is that when the second call to `ReceiveAsync` happens, the second call sets up the exception on the TCS returned by the _first_ call, then also returns that same task:
```
if (receiveOperationAlreadyPending)
{
    var exception = new InvalidOperationException(
    SR.Format(SR.net_Websockets_AlreadyOneOutstandingOperation, "ReceiveAsync"));
    _operation.TcsReceive.TrySetException(exception);

    Abort();
}

return _operation.TcsReceive.Task;
```
(Given the locking that occurs, this is guaranteed to happen even if submitted on two concurrently running threads, although which one gets it would be up in the air)

This is clearly different from the behavior on Linux/elsewhere, when going through [ManagedWebSocket](https://github.com/dotnet/corefx/blob/master/src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ManagedWebSocket.cs#L288-L309) it returns a separate task based on a thrown exception, leaving the first task to complete "on it's own" (shutdown gracefully, IIRC, since the connection ends up closed):
```
try
{
    WebSocketValidate.ThrowIfInvalidState(_state, _disposed, s_validReceiveStates);

    Debug.Assert(!Monitor.IsEntered(StateUpdateLock), $"{nameof(StateUpdateLock)} must never be held when acquiring {nameof(ReceiveAsyncLock)}");
    lock (ReceiveAsyncLock) // synchronize with receives in CloseAsync
    {
        ThrowIfOperationInProgress(_lastReceiveAsync);
        Task<WebSocketReceiveResult> t = ReceiveAsyncPrivate(buffer, cancellationToken);
        _lastReceiveAsync = t;
        return t;
        }
}
catch (Exception exc)
{
    return Task.FromException<WebSocketReceiveResult>(exc);
}
```
... soooo now what?  Even without knowing what's going on inside, I don't feel like I would describe this as a race condition, looking at it externally (Assuming that there's no chance for the call to resolve first).

Is this something we should special case out based on implementation used, or should one of the versions be changed to match the other?